pkg/controller/config-controller.go

```diff
@@ -0,0 +1,384 @@
+			config = conf
+		}
+	}
+
+	if config == nil {
+		runtime.HandleError(errors.Errorf("error getting CDI config"))
+		return
+	}
+
+	if ing, ok := obj.(*extensionsv1beta1.Ingress); ok {
+		for _, rule := range ing.Spec.Rules {
+			for _, path := range rule.HTTP.Paths {
+				if path.Backend.ServiceName == c.uploadProxyServiceName {
+					c.enqueueCDIConfig(config)
+					return
+				}
+			}
+		}
+	}
+	if route, ok := obj.(*routev1.Route); ok {
+		if route.Spec.To.Name == c.uploadProxyServiceName {
+			c.enqueueCDIConfig(config)
+			return
+		}
+	}
+	if conf, ok := obj.(*cdiv1.CDIConfig); ok {
+		if conf.Name == config.Name {
+			c.enqueueCDIConfig(conf)
+			return
+		}
+	}
+
+	return
+}
+
+func (c *ConfigController) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *ConfigController) processNextWorkItem() bool {
+	obj, shutdown := c.queue.Get()
+
+	if shutdown {
+		return false
+	}
+
+	err := func(obj interface{}) error {
+		defer c.queue.Done(obj)
+
+		var key string
+		var ok bool
+
+		if key, ok = obj.(string); !ok {
+			c.queue.Forget(obj)
+			runtime.HandleError(errors.Errorf("expected string in workqueue but got %#v", obj))
+			return nil
+		}
+
+		if err := c.syncHandler(key); err != nil {
+			return errors.Errorf("error syncing '%s': %s", key, err.Error())
+		}
+
+		c.queue.Forget(obj)
+		glog.Infof("Successfully synced '%s'", key)
+		return nil
+
+	}(obj)
+
+	if err != nil {
+		runtime.HandleError(err)
+		return true
+	}
+
+	return true
+}
+
+func (c *ConfigController) syncHandler(key string) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(errors.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	var config *cdiv1.CDIConfig
+	configs, err := c.configLister.CDIConfigs(namespace).List(labels.NewSelector())
+	if err != nil {
+		return err
+	}
+	for _, conf := range configs {
+		if conf.Name == name {
+			config = conf
+		}
+	}
+
+	if config == nil {
+		runtime.HandleError(errors.Errorf("CDIConfig '%s' in work queue no longer exists", key))
+		return nil
+	}
+
+	var url string
+	if config.Spec.UploadProxyURLOverride != nil {
+		if (*config.Spec.UploadProxyURLOverride) == (*config.Status.UploadProxyURL) {
+			//override is already set and status reflects it
+			return nil
+		}
+		url = *config.Spec.UploadProxyURLOverride
+	} else {
+
+		routes, err := c.routeLister.List(labels.NewSelector())
+		if err != nil {
+			return err
+		}
+		for _, route := range routes {
+			url = getURLFromRoute(route, c.uploadProxyServiceName)
+			if url != "" {
+				break
+			}
+		}
+		if url == "" {
+			ingresses, err := c.ingressLister.List(labels.NewSelector())
+			if err != nil {
+				return err
+			}
+			for _, ing := range ingresses {
+				url = getURLFromIngress(ing, c.uploadProxyServiceName)
+				if url != "" {
+					break
+				}
+			}
+		}
+	}
+
+	newConfig := config.DeepCopy()
+	// mutate newConfig
+	newConfig.Status.UploadProxyURL = &url
```

Comment:
looks like it is possible to get here and have `url == ""`.  If so, `newConfig.Status.UploadProxyURL` should be `nil`.  Not the address of `url`.

Also, if the url did not change, should we still do an update?